internal/clock/simulated_clock.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clock
+
+import (
+	"sync"
+	"time"
+)
+
+// afterRequest holds the information for a pending After call in SimulatedClock.
+type afterRequest struct {
+	targetTime time.Time
+	ch         chan time.Time
+}
+
+// SimulatedClock is a clock that allows for manipulation of the time,
+// which does not change unless AdvanceTime or SetTime is called.
+// The zero value is a clock initialized to the zero time.
+type SimulatedClock struct {
+	mu      sync.RWMutex
+	t       time.Time       // GUARDED_BY(mu)
+	pending []*afterRequest // GUARDED_BY(mu)
+}
+
+func NewSimulatedClock(startTime time.Time) *SimulatedClock {
+	return &SimulatedClock{
+		t:       startTime,
+		pending: make([]*afterRequest, 0),
```

Comment:
Why not start with a nil value?
```suggestion
		pending: nil,
```